src/main/java/net/snowflake/client/jdbc/telemetryOOB/TelemetryService.java

```diff
@@ -142,6 +164,10 @@ public boolean isHTAPEnabled() {
     }
   }
 
+  public boolean isTelemetryEnabled() {
```

Comment:
should it be `isInBoundTelemetryEnabled`?

Reply:
But it does not seems to be right that on the session parameter we are changing the global state - but it's already merged for a long time

Nevertheless. For sending the new telemetry events you are adding we should be do this via inbound telemetry?

Reply:
Yes, it should go to to client_telemetry_v with the session and query_id. There are already some fields that are sent, we are trying to add more.